### PR TITLE
[MAINTENANCE] Other dialect test

### PR DESCRIPTION
**Pull request by anthonyburdi**
Changes proposed in this pull request:
- Test that an OTHER dialect is appropriately handled by `SqlalchemyBatchData`
- End to end test retrieving data with OTHER dialect
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in github issues or slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/execution_engine/test_sqlalchemy_batch_data.py [hunk 1]
```diff
@@ -134,3 +135,16 @@ def test_instantiation_with_and_without_temp_table(sqlite_view_engine, sa):
     )
     res = execution_engine.get_batch_data_and_markers(batch_spec=my_batch_spec)
     assert len(res) == 2
+
+
+def test_instantiation_with_unknown_dialect(sqlite_view_engine):
```
Comment (donaldheppner) [inline tests/execution_engine/test_sqlalchemy_batch_data.py:141]: I think unsupported is the wrong word here; "other" would suffice.

**Review by donaldheppner — CHANGES_REQUESTED** [review]

File: tests/execution_engine/test_sqlalchemy_batch_data.py (hunk 1 above)
Comment (anthonyburdi) [reply tests/execution_engine/test_sqlalchemy_batch_data.py:141]: Done!

Comment (billdirks) [inline tests/execution_engine/test_sqlalchemy_batch_data.py:141]: Could you mark this test with `@pytest.mark.unit`

File: tests/execution_engine/test_sqlalchemy_dialect.py [hunk 2]
```diff
@@ -9,9 +9,9 @@ def test_dialect_instantiation_with_byte_string():
     assert GESqlDialect(b"hive") == GESqlDialect.HIVE
 
 
-def test_get_all_dialect_names_no_unsupported():
-    assert GESqlDialect.UNSUPPORTED.value not in GESqlDialect.get_all_dialect_names()
+def test_get_all_dialect_names_no_other_dialects():
```
Comment (billdirks) [inline tests/execution_engine/test_sqlalchemy_dialect.py:17]: Similarly for these tests: @pytest.mark.unit

File: tests/execution_engine/test_sqlalchemy_execution_engine.py [hunk 3]
```diff
@@ -69,6 +71,42 @@ def test_instantiation_via_url(sa):
...
+        batch_spec=SqlAlchemyDatasourceBatchSpec(
+            table_name="table_partitioned_by_date_column__A",
+            sampling_method="_sample_using_limit",
+            sampling_kwargs={"n": num_rows_in_sample},
+        )
+    )
+
+    # 4. Assert dialect and data are as expected
+
+    assert batch_data.dialect == GESqlDialect.OTHER
```
Comment (billdirks) [inline tests/execution_engine/test_sqlalchemy_execution_engine.py:103]: 🚀

**Review by billdirks — APPROVED** [review]
Could you mark some of the new tests with `@pytest.mark.unit`? 

Otherwise LGTM!

**Review by donaldheppner — APPROVED** [review]

**Pull request merged**